src/python/pants/engine/streaming_workunit_handler.py

```diff
@@ -198,14 +224,24 @@ def __init__(
         callbacks: Iterable[WorkunitsCallback],
         report_interval: float,
         max_workunit_verbosity: LogLevel,
-    ):
+    ) -> None:
         super().__init__(daemon=True)
         self.scheduler = scheduler
         self._context = context
         self.stop_request = threading.Event()
         self.report_interval = report_interval
         self.callbacks = callbacks
         self.max_workunit_verbosity = max_workunit_verbosity
+        # Get the parent thread's logging destination. Note that this thread has not yet started
+        # as we are only in the constructor.
+        self.logging_destination = native_engine.stdio_thread_get_destination()
+
+    def _maybe_set_logging_destination(self) -> None:
+        """Set the thread's logging destination to the parent thread's, meaning the console."""
+        if self.logging_destination is None:
+            return
+        native_engine.stdio_thread_set_destination(self.logging_destination)
+        self.logging_destination = None
```

Comment:
Because this type has an `Arc` behind it (and because all Python objects are effectively "arc'd" already), there is no need to do this one-time-use thing. Can pass around as many references as you'd like.

So probably also safe to inline the get/set, IMO.

Reply:
Ah, my thinking was instead that `run()` would execute `native_engine.stdio_thread_set_destination` multiple times in a loop. But I see that the loop is only because we have an explicit `while` loop. Will inline.